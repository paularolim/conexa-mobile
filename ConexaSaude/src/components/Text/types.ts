export type { TextProps } from 'react-native';
