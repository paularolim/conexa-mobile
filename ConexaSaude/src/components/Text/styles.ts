import { Text } from 'react-native';
import styled from 'styled-components/native';

export const RNText = styled(Text)`
  color: ${({ theme }) => theme.colors.black};
`;
