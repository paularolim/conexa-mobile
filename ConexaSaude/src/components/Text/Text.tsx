import { RNText } from './styles';
import { TextProps } from './types';

export function Text({ children, ...rest }: TextProps) {
  return <RNText {...rest}>{children}</RNText>;
}
