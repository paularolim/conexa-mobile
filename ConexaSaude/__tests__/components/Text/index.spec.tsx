import renderer from 'react-test-renderer';

import { Text } from '@components/Text';

import { Wrapper } from '../../Wrapper';

describe('Text component', () => {
  it('should render default component correctly', () => {
    const tree = renderer
      .create(
        <Wrapper>
          <Text>Button component</Text>
        </Wrapper>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
