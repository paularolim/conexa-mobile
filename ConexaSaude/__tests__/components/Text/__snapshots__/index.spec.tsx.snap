// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component should render default component correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#000000",
      },
    ]
  }
>
  Button component
</Text>
`;
